### DISCUSSION TITLE
blueman-manager starts after D-Bus timeout when starting from blueman-applet

### DISCUSSION THREAD

**Discussion by felixfrank** [blueman-project/blueman#2356] [Q&A]
blueman: 2.4.1
BlueZ: current git build 5.75
Distribution: archlinux kernel-6.8.9
Desktop environment: i3

* [x] I have consulted the [Troubleshooting page](https://github.com/blueman-project/blueman/wiki/Troubleshooting) and done my best effort to follow.

When I try to start `blueman-manager` from the `blueman-applet` by double clicking on the tray icon, the manager window takes ~25 seconds to start, whereas it starts immediately when started from other means. For example from the cmdline.
In the `blueman-applet` log I get the following lines:
``` shell
blueman-applet 14.48.51 ERROR    DBusProxies:62 activate  : Call to org.blueman.Manager failed
Traceback (most recent call last):
  File "/usr/lib/python3.12/site-packages/blueman/main/DBusProxies.py", line 57, in activate
    self.call_sync("Activate", param, Gio.DBusCallFlags.NONE, -1, None)
gi.repository.GLib.GError: g-io-error-quark: Timeout was reached (24)
Traceback (most recent call last):
  File "/usr/lib/python3.12/site-packages/blueman/main/indicators/GtkStatusIcon.py", line 64, in <lambda>
    self.indicator.connect('activate', lambda _status_icon: tray.activate_status_icon())
                                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/blueman/main/Tray.py", line 65, in activate_status_icon
    AppletService().Activate()
  File "/usr/lib/python3.12/site-packages/gi/overrides/Gio.py", line 349, in __call__
    result = self.dbus_proxy.call_sync(self.method_name, arg_variant,
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
gi.repository.GLib.GError: g-io-error-quark: Timeout was reached (24)
```

After the timeout I get a notification saying:
``` shell
Failed to reach blueman-manager

It seems like blueman-manager could no get activated via D-Bus.
A typical cause for this is a broken graphical setup in the D-Bus activation environment
that can get resolved with a call to dbus-update-activation-environment,
typically issued from xinitrc (respectively the Sway config or similar).
```

I did try calling `dbus-update-activation-environment` before starting the `blueman-applet` with no effect.

Here is the [full `blueman-applet` log](https://pastebin.com/4wNmK6G6).

**Comment by infirit** [comment]
You need to update your environment as early as possible in your i3 configuration with the appropriate arguments. I don't use i3 so you might want to ask on their support channels how to properly configure your environment.
